exchanges/sharedtestvalues/sharedtestvalues.go

```diff
@@ -47,3 +55,48 @@ func NewTestWebsocket() *stream.Websocket {
 		Match:             stream.NewMatch(),
 	}
 }
+
+// SkipUnsetCredentials is a test helper function checking if the authenticated function
+// can perform the required test.
+func SkipUnsetCredentials(t *testing.T, exch exchange.IBotExchange, canManipulateOrders ...bool) {
+	t.Helper()
+
+	areTestAPIKeysSet := AreAPIkeysSet(exch)
+	supportsManipulatingOrders := len(canManipulateOrders) > 0
+	allowedToManipulateOrders := supportsManipulatingOrders && canManipulateOrders[0]
+
+	if areTestAPIKeysSet && !supportsManipulatingOrders ||
+		areTestAPIKeysSet && allowedToManipulateOrders {
+		return
+	}
+
+	message := []string{warningSkip}
+	if !areTestAPIKeysSet {
+		message = append(message, warningKeys)
+	}
+
+	if supportsManipulatingOrders && !allowedToManipulateOrders {
+		message = append(message, warningManipulateOrders)
+	}
+	message = append(message, warningHowTo)
+	t.Skip(strings.Join(message, ", "))
+}
+
+// SkipCredentialsSetCantManipulate will only skip if the credentials are set
+// correctly and can manipulate orders is set to false. It will continue normal
+// operations if credentials are not set, giving better code coverage.
+func SkipCredentialsSetCantManipulate(t *testing.T, exch exchange.IBotExchange, canManipulateOrders bool) {
+	t.Helper()
+
+	if !AreAPIkeysSet(exch) || canManipulateOrders {
+		return
+	}
+
+	message := []string{warningSkip, warningManipulateOrders}
+	t.Skip(strings.Join(message, ", "))
+}
+
+// AreAPIkeysSet returns if the API keys are set.
+func AreAPIkeysSet(exch exchange.IBotExchange) bool {
```

Comment:
You interchange between credentials and API keys. Let's keep them the same, either way: only credentials or only APIKeys